### PR TITLE
VTK-m: Point to github mirror for source tarball

### PR DESCRIPTION
**Pull request by kwryankrattiger**
The gitlab.kitware.com source location seems to have intermittent network issues. Switching the to mirror hosted at Github may alleviate some of the timeouts.

CC: @vicentebolea

### CODE REVIEW

**Comment by kmorel** [comment]
It looks like the tarballs provided by GitHub have different SHAs than those provided by GitLab. Here is the output when I tried to install vtk-m@2.1.0 on Frontier:

```
==> Installing vtk-m-2.1.0-so2h2eiixjcc5knaxatfntx64vxgn5s4 [50/50]
==> No binary for vtk-m-2.1.0-so2h2eiixjcc5knaxatfntx64vxgn5s4 found: installing from source
==> Fetching https://github.com/Kitware/VTK-m/archive/refs/tags/v2.1.0.tar.gz
==> Error: ChecksumError: sha256 checksum failed for /tmp/kmorel/spack-stage/spack-stage-vtk-m-2.1.0-so2h2eiixjcc5knaxatfntx64vxgn5s4/v2.1.0.tar.gz
    Expected 9cf3522b6dc0675281a1a16839464ebd1cc5f9c08c20eabee1719b3bcfdcf41f but got 7b224f1f91e5ef140e193338bf091133b1e9f40d323bccdc8bb80bfc2675e6ea. File size = 3208035 bytes. Contents = b'\x1f\x8b\x08\x00\x00\x00\x00\x00\x00\x03\xec=ks\xda\xc8...WuU\xd7w\xbb\xfe\x07U$\x90\xf2\x00\xf83\x01'
```

Likewise, here is @2.2.0:

```
==> Installing vtk-m-2.2.0-owxax4oib2pzzvohaoqwzy64vqwlhfx6 [28/28]
==> No binary for vtk-m-2.2.0-owxax4oib2pzzvohaoqwzy64vqwlhfx6 found: installing from source
==> Fetching https://github.com/Kitware/VTK-m/archive/refs/tags/v2.2.0.tar.gz
==> Error: ChecksumError: sha256 checksum failed for /tmp/kmorel/spack-stage/spack-stage-vtk-m-2.2.0-owxax4oib2pzzvohaoqwzy64vqwlhfx6/v2.2.0.tar.gz
    Expected ee66b6bbd33f6ad6f2350e11a7c9328492e53935ba8f66b4b1d01f074eb96341 but got f40d6b39ca1bcecd232571c92ce606627811909f4e21972d1823e605f686bcf5. File size = 3529567 bytes. Contents = b'\x1f\x8b\x08\x00\x00\x00\x00\x00\x00\x03\xec=\xfdw\xda\xb8...JV\xb2\xf2\x87\x95\xff\x01h9\x9bq\x00\x00O\x01'
```

I think all the SHAs need to be updated.

On the plus side, I didn't get any timeouts.

**Comment by zackgalbreath** [comment]
Evidence that my latest push correctly updated sha256 for fetching from GitHub:

```
$ spack fetch vtk-m@2.2.0
==> Fetching https://github.com/Kitware/VTK-m/archive/refs/tags/v2.2.0.tar.gz
$ spack fetch vtk-m@2.1.0
==> Fetching https://github.com/Kitware/VTK-m/archive/refs/tags/v2.1.0.tar.gz
$ spack fetch vtk-m@2.0.0
==> Fetching https://github.com/Kitware/VTK-m/archive/refs/tags/v2.0.0.tar.gz
$ spack fetch vtk-m@1.9.0
==> Fetching https://github.com/Kitware/VTK-m/archive/refs/tags/v1.9.0.tar.gz
$ spack fetch vtk-m@1.8.0
==> Fetching https://github.com/Kitware/VTK-m/archive/refs/tags/v1.8.0.tar.gz
$ spack fetch vtk-m@1.7.1
==> Fetching https://github.com/Kitware/VTK-m/archive/refs/tags/v1.7.1.tar.gz
$ spack fetch vtk-m@1.7.0
==> Fetching https://github.com/Kitware/VTK-m/archive/refs/tags/v1.7.0.tar.gz
$ spack fetch vtk-m@1.6.0
==> Fetching https://github.com/Kitware/VTK-m/archive/refs/tags/v1.6.0.tar.gz
$ spack fetch vtk-m@1.5.1
==> Fetching https://github.com/Kitware/VTK-m/archive/refs/tags/v1.5.1.tar.gz
$ spack fetch vtk-m@1.5.0
==> Fetching https://github.com/Kitware/VTK-m/archive/refs/tags/v1.5.0.tar.gz
$ spack fetch vtk-m@1.4.0
==> Fetching https://github.com/Kitware/VTK-m/archive/refs/tags/v1.4.0.tar.gz
$ spack fetch vtk-m@1.3.0
==> Fetching https://github.com/Kitware/VTK-m/archive/refs/tags/v1.3.0.tar.gz
$ spack fetch vtk-m@1.2.0
==> Fetching https://github.com/Kitware/VTK-m/archive/refs/tags/v1.2.0.tar.gz
$ spack fetch vtk-m@1.1.0
==> Fetching https://github.com/Kitware/VTK-m/archive/refs/tags/v1.1.0.tar.gz
```

**Review by vicentebolea — APPROVED** [review]
Thanks for taking care of this. IMHO this should be temporal until we resolve this get timeout issue

**Review by kmorel — APPROVED** [review]
I also verify that the updated SHAs allow the correct downloads.

File: var/spack/repos/builtin/packages/vtk-m/package.py [hunk 1]
```diff
@@ -31,26 +31,26 @@ class VtkM(CMakePackage, CudaPackage, ROCmPackage):
     version("release", branch="release")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/vtk-m/package.py:31]: There doesn't appear to be a `release` branch.  Do you want to add it or remove the `version` directive here?

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Confirmed all of the sha256s, commits, and one of the two branches.  Let me know what you want to do about the `releases` branch directive.

File: var/spack/repos/builtin/packages/vtk-m/package.py (hunk 1 above)
Comment (kmorel) [reply var/spack/repos/builtin/packages/vtk-m/package.py:31]: Hmm. How hard would it be to have the `release` branch mirrored on GitHub?

Comment (zackgalbreath) [reply var/spack/repos/builtin/packages/vtk-m/package.py:31]: I'm looking into this.

Comment (zackgalbreath) [reply var/spack/repos/builtin/packages/vtk-m/package.py:31]: Ok, we should be syncing the `release` branch now too.

https://github.com/Kitware/VTK-m/tree/release

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/vtk-m/package.py:31]: Works for me.  Thanks!

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**